rust/libnewsboat/src/utils.rs

```diff
@@ -265,8 +267,78 @@ pub fn unescape_url(rs_str: String) -> Option<String> {
     Some(result.unwrap().replace("\0",""))
 }
 
+pub fn get_command_output(cmd: &str) -> String {
+    let cmd = Command::new("sh").arg("-c").arg(cmd).output();
+    // from_utf8_lossy will convert any bad bytes to U+FFFD
+    cmd.map(|cmd| String::from_utf8_lossy(&cmd.stdout).into_owned())
+        .unwrap_or_else(|_| String::from(""))
+}
+
+pub fn run_command(cmd: &str, param: &str) {
+    let child = Command::new(cmd)
+        .arg(param)
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn();
+    if let Err(error) = child {
+        log!(Level::Debug,
+             &format!("utils::run_command: spawning a child for \"{}\" failed: {}", cmd, error));
+    }
+
+    /* We deliberately *don't* wait for the child to finish. */
+}
+
+pub fn run_program(cmd_with_args: &[&str], input: &str) -> String {
+    if cmd_with_args.is_empty() {
+        return String::new();
+    }
+
+    let child = Command::new(cmd_with_args[0])
+        .args(&cmd_with_args[1..])
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::null())
+        .spawn();
+
+    match child {
```

Comment:
This is probably a matter of taste, but I find Result combinators easier to read than match. See https://github.com/rnestler/newsboat/pull/new/get-command-output-review

Reply:
I ruled them out without even trying, because it seemed like they won't improve the code much. Your code proves me wrong; I'll use it. Thank you!